picard/ui/scripteditor.py

```diff
@@ -448,44 +611,102 @@ def toggle_wordwrap(self):
+            msg (str): Error message to display
+        """
+        log.error(fmt, filename, msg)
+        error_message = _(fmt) % (filename, _(msg))
+        self.display_error(ScriptImportError(_(title), error_message))
+
+    def output_file_error(self, fmt, filename, msg):
+        """Log file error and display error message dialog.
+
+        Args:
+            fmt (str): Format for the error type being displayed
+            filename (str): Name of the file being imported or exported
+            msg (str): Error message to display
+        """
+        self.output_error(_("File Error"), fmt, filename, msg)
+
     def import_script(self):
-        """Import the current script from an external text file.
+        """Import from an external text file to a new script. Import can be either a plain text script or
+        a naming script package.
         """
-        dialog_title = N_("Import Script File")
-        dialog_file_types = self.FILE_TYPE_SCRIPT + ";;" + self.FILE_TYPE_ALL
+
+        dialog_title = _("Import Script File")
+        dialog_file_types = self.FILE_TYPE_PACKAGE + ";;" + self.FILE_TYPE_SCRIPT + ";;" + self.FILE_TYPE_ALL
         options = QtWidgets.QFileDialog.Options()
         options |= QtWidgets.QFileDialog.DontUseNativeDialog
         filename, file_type = QtWidgets.QFileDialog.getOpenFileName(self, dialog_title, self.default_script_directory, dialog_file_types, options=options)
         if filename:
             log.debug('Importing naming script file: %s' % filename)
             try:
                 with open(filename, 'r', encoding='utf8') as i_file:
-                    self.set_script(i_file.read())
+                    file_content = i_file.read()
             except OSError as error:
-                error_message = 'Error importing file "{0}": {1}'.format(filename, error.strerror)
-                log.error(error_message)
-                self.display_error(OptionsCheckError(N_("File Error"), error_message))
+                self.output_file_error(self.FILE_ERROR_IMPORT, filename, error.strerror)
+                return
+            if not file_content.strip():
+                self.output_file_error(self.FILE_ERROR_IMPORT, filename, N_('The file was empty'))
+                return
+            if file_type == self.FILE_TYPE_PACKAGE:
+                try:
+                    script_item = FileNamingScript().create_from_json(file_content)
+                except JSONDecodeError as error:
+                    self.output_file_error(self.FILE_ERROR_DECODE, filename, error.msg)
+                    return
+                if not (script_item.get_value('title') and script_item.get_value('script')):
+                    self.output_file_error(self.FILE_ERROR_DECODE, filename, N_('Invalid script package'))
+                    return
+            else:
+                script_item = FileNamingScript(
+                    title=_("Imported from ") + filename,
```

Comment:
Avoid concatenation for i18n.

```suggestion
                    title=_("Imported from %s") % filename,
```